x-pack/plugins/siem/server/lib/events/query.last_event_time.dsl.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { LastEventTimeRequestOptions } from './types';
+
+interface EventIndicies {
+  [key: string]: string[];
+}
+
+export const buildLastEventTimeQuery = ({
+  indexKey,
+  details,
+  sourceConfiguration: {
+    fields: { timestamp },
+    logAlias,
+    auditbeatAlias,
+    packetbeatAlias,
+    winlogbeatAlias,
+  },
+}: LastEventTimeRequestOptions) => {
+  const indicesToQuery: EventIndicies = {
+    hosts: [logAlias, auditbeatAlias, packetbeatAlias, winlogbeatAlias],
+    network: [logAlias, packetbeatAlias],
+  };
+  const getHostDetailsFilter = (hostName: string) => [{ term: { 'host.name': hostName } }];
+  const getIpDetailsFilter = (ip: string) => [
+    { term: { 'source.ip': ip } },
+    { term: { 'destination.ip': ip } },
+  ];
+  switch (indexKey) {
```

Comment:
Optional: If you add a type for `indexKey` to `LastEventTimeProps`, you can then take advantage of [exhaustive](http://ideasintosoftware.com/exhaustive-switch-in-typescript/) [checking](https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking) and get a type assertion if a new `indexKey` is added. Search for `assertUnreachable()` for a few examples throughout the code base.